shared/src/main/java/net/skinsrestorer/shared/utils/LocaleParser.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SkinsRestorer
+ *
+ * Copyright (C) 2022 SkinsRestorer
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ */
+package net.skinsrestorer.shared.utils;
+
+import net.skinsrestorer.shared.storage.Config;
+
+import java.util.Locale;
+
+public class LocaleParser {
+    public static Locale parseLocale(String locale) {
+        if (locale == null) {
+            return Config.LANGUAGE;
+        }
+        String[] split = locale.split("_");
+        if (split.length == 1) {
+            return new Locale(split[0]);
+        } else if (split.length == 2) {
+            return new Locale(split[0], split[1]);
+        } else {
+            return Config.LANGUAGE;
+        }
+    }
+
+    public static Locale parseLocaleStrict(String locale) {
+        String[] split = locale.split("_");
+        if (split.length == 1) {
+            return new Locale(split[0]);
+        } else if (split.length == 2) {
+            return new Locale(split[0], split[1]);
+        }
+        throw new IllegalArgumentException("Invalid locale: " + locale);
```

Comment:
add more info on how to resolve / why it's invalid (example using " ' mixup and how to escape)

Reply:
It's just about the format it is in. I'll add a notice.